generator/src/main/java/com/google/archivepatcher/generator/FileByFileDeltaGenerator.java

```diff
@@ -118,4 +126,12 @@ private PreDiffPlan generatePreDiffPlan(
   protected DeltaGenerator getDeltaGenerator() {
     return new BsDiffDeltaGenerator();
   }
+
+  private static <T> List<T> getImmutableCopy(List<T> input) {
```

Comment:
Same comment as in pull 32, this isn't making an immutable copy, but rather an unmodifiable wrapper around a list of undetermined mutability.

Reply:
Yup. I forgot to add `new ArrayList<>()`. A bit messed up in between the CLs. 

Btw, we are considering using guava immutable lists. I heard that there was a conscious choice to stay away from guava for open source compatibility. Considering Guava is supported in Gradle now, WDYT?

Reply:
It was not so much about Gradle and more about pulling in the entire library in general.